tests/e2e/helm.go

```diff
@@ -0,0 +1,294 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"emperror.dev/errors"
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo/v2"
+	"sigs.k8s.io/yaml"
+)
+
+// HelmDescriptor describes a component that can be operated on by Helm.
+// Repository+ChartName and ChartPath are mutually exclusive.
+type helmDescriptor struct {
+	Repository                   string
+	ChartName                    string
+	ChartVersion                 string
+	ReleaseName                  string
+	Namespace                    string
+	SetValues                    map[string]string
+	HelmExtraArguments           map[string][]string
+	RemoteCRDPathVersionTemplate string
+	LocalCRDSubpaths             []string
+	LocalCRDTemplateRenderValues map[string]string
+}
+
+// crdPath returns the path of the CRD belonging to the Helm descriptor based on
+// the chart version and local/remote Helm chart.
+func (helmDescriptor *helmDescriptor) crdPath() (string, error) {
+	if helmDescriptor == nil {
+		return "", errors.Errorf("invalid nil Helm descriptor")
+	}
+
+	if helmDescriptor.IsRemote() {
+		return fmt.Sprintf(
+			helmDescriptor.RemoteCRDPathVersionTemplate,
+			strings.TrimPrefix(helmDescriptor.ChartVersion, "v"),
+		), nil
+	}
+
+	localCRDsBytes := []byte(helm.RenderTemplate(
+		GinkgoT(),
+		&helm.Options{
+			SetValues: helmDescriptor.LocalCRDTemplateRenderValues,
+		},
+		helmDescriptor.Repository,
+		helmDescriptor.ReleaseName,
+		[]string{"templates/crds.yaml"},
+	))
+
+	return createTempFileFromBytes(localCRDsBytes, "", "", 0)
+}
+
+// installHelmChart checks whether the specified named Helm release exists in
+// the provided kubectl context and namespace, logs it if it does and returns or
+// alternatively deploys a Helm chart to the specified kubectl context and
+// namespace using the specified infos, extra arguments can be any of the helm
+// CLI install flag arguments, flag keys and values must be provided separately.
+func (helmDescriptor *helmDescriptor) installHelmChart(kubectlOptions k8s.KubectlOptions) error {
+	if helmDescriptor == nil {
+		return errors.Errorf("invalid nil Helm descriptor")
+	}
+
+	kubectlOptions.Namespace = helmDescriptor.Namespace
+
+	if !helmDescriptor.IsRemote() { // Note: local chart with directory path in helmDescriptor.Repository.
```

Comment:
Not crucial in any way:
```suggestion
	if !helmDescriptor.IsRemote() { // Note: local chart with directory path in helmDescriptor.Repository
```

Reply:
Are you asking to remove the sentence terminating dot?

My reasoning for adding it was the [comment sentences](https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences) Go idiom.

I can do so easily, just want to clarify what we aim for - so far I tried to uphold that idiom.

Reply:
Ah I see, then just ignore this, didn't know about that.